runtime/src/main/kotlin/kotlin/text/Strings.kt

```diff
@@ -177,8 +177,10 @@ public actual external fun String.toLowerCase(): String
 /**
  * Returns a new character array containing the characters from this string.
  */
+public fun String.toCharArray(): CharArray = toCharArray(this, 0, length)
+
 @SymbolName("Kotlin_String_toCharArray")
-public external fun String.toCharArray(): CharArray
+private external fun toCharArray(string: String, start: Int, size: Int): CharArray
```

Comment:
Why?

Reply:
Now it's a part of a function without bound checking.